projects/lib/src/board/board.h

```diff
@@ -325,8 +325,12 @@ class LIB_EXPORT Board
 				  const QString& symbol,
 				  unsigned movement = 0,
 				  const QString & gsymbol = QString());
+		/*! Returns true if \a piece on \a square can capture like \a movement. */
+		bool pieceHasCaptureMovement(Piece piece, int square, unsigned movement) const;
+		/*! Returns true if \a piece on \a square can move like \a movement. */
+		bool pieceHasMovement(Piece piece, int square, unsigned movement) const;
 		/*! Returns true if \a pieceType can move like \a movement. */
-		bool pieceHasMovement(int pieceType, unsigned movement) const;
+		bool pieceTypeHasMovement(int pieceType, unsigned movement) const;
```

Comment:
Is it intentional that this is not a virtual method (it is reimplemented in derived classes)?

Reply:
This was unintentional.  I think you are correct that adding the `virtual` keyword would make sense here (I don't recall semantic differences between virtual and pure virtual functions, but I think you are correct that `virtual` is semantically appropriate here).

Reply:
It's actually very important to use virtual functions instead of overridden (non-virtual) functions. We're using pointers to `Chess::Board` pretty much everywhere, and without the `virtual` keyword any `board->pieceHasMovement()` calls would go to `Chess::Board::pieceHasMovement()` when the intention is to call the reimplemented method.